wgengine/router/ifconfig_windows.go

```diff
@@ -588,28 +640,46 @@ func deltaRouteData(a, b []*winipcfg.RouteData) (add, del []*winipcfg.RouteData)
 	return
 }
 
+// getInterfaceRoutes returns all the interface's routes.
+// Corresponds to GetIpForwardTable2 function, but filtered by interface.
+func getInterfaceRoutes(ifc *winipcfg.IPAdapterAddresses, family winipcfg.AddressFamily) ([]*winipcfg.MibIPforwardRow2, error) {
+	routes, err := winipcfg.GetIPForwardTable2(family)
+	if err != nil {
+		return nil, err
+	}
+	matches := make([]*winipcfg.MibIPforwardRow2, len(routes))
+	i := 0
+	for _, route := range routes {
+		if route.InterfaceLUID == ifc.LUID {
+			matches[i] = &route
```

Comment:
routes is of type `[]T` (https://godoc.org/golang.zx2c4.com/wireguard/windows/tunnel/winipcfg#GetIPForwardTable2) not `[]*T`, and this is a range over that value type, so `&route` is the same value as you iterate as you range over the slice.

Reply:
Oops. Fixed.

I was doing this change pretty much mechanically, without much understanding of what I was looking at. Go compiler was guiding me along. You can blame the compiler.

PTAL

Alex